### ISSUE TITLE
E/REALM: uncaught exception in notifier thread: N5realm15InvalidTableRefE: transaction_ended

### ISSUE THREAD

**Issue by 2u** [realm/realm-java#7250]
#### Goal
We can't understand and locate problem
at all

#### Actual Results
E/REALM: uncaught exception in notifier thread: N5realm15InvalidTableRefE: transaction_ended
    Exception backtrace:
    <backtrace not supported on this platform>
    
    
    --------- beginning of crash
A/libc: /usr/local/google/buildbot/src/android/ndk-release-r21/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:72: abort_message: assertion "terminating with uncaught exception of type realm::InvalidTableRef: transaction_ended
    Exception backtrace:
    <backtrace not supported on this platform>" failed
    Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 21101 , pid 21050 
### 

#### Steps & Code to Reproduce
exception crashes the app at unpredictable time
it seems it does not depends on code (we can't find any correlations)
all our code was tested and runs fully without logical errors\exceptions
and then suddenly app crashes
it happen after 10-20 sec of testing or may be after 20 minutes

#### Version of Realm and tooling

Realm version(s): 10.2

Realm Sync feature enabled: No

Android Studio version: 4.1

Android Build Tools version: 30.0.3

Gradle version: 6.5
Gradle plugin version: 4.1.1

Which Android version and device(s): Pixel_2_API_29 (emulator, osX and windows)

**Comment by tgoyne** [comment]
I coincidentally just hit this while working on something unrelated and I think I see what's going wrong.

**Comment by tgoyne** [comment]
Probably will be fixed by https://github.com/realm/realm-core/pull/4234.

**Comment by cmelchior** [comment]
There is a chance this might be fixed by Realm Java ` 10.3.0`. Which will be released shortly. At least this was in the changelog: https://github.com/realm/realm-core/blob/master/CHANGELOG.md#1033-release-notes

```
Fix a race condition which would lead to "uncaught exception in notifier thread: N5realm15InvalidTableRefE: transaction_ended" and a crash when the source Realm was closed or invalidated at a very specific time during the first run of a collection notifier (#3761, since v6.0.0).
```

**Comment by rorbech** [comment]
@2u Closing as considered fixed in 10.3.0. Please reopen if you are still seeing this with 10.3.0.
